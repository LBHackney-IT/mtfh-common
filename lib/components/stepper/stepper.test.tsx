import React from "react";
import { render, testA11y } from "@hackney/mtfh-test-utils";
import { screen } from "@testing-library/react";
import { Stepper } from "./Stepper";

const stepperTitle = "Stepper title";
const stepTenureDetails = "Tenure details";
const stepSelectResidents = "Select residents";

test("it renders correctly", async () => {
  const { container } = render(
    <Stepper
      title={stepperTitle}
      steps={[stepTenureDetails, stepSelectResidents]}
      activeStep={0}
    />,
  );
  expect(container).toMatchSnapshot();
  expect(screen.queryAllByRole("listitem").length).toBe(2);
  expect(screen.queryAllByRole("listitem")[0].className).toContain("active");
  await testA11y(container);
});

test("it prints a title", () => {
  render(
    <Stepper title={stepperTitle} steps={[stepTenureDetails, stepSelectResidents]} />,
  );
  screen.getByText(stepperTitle);
});

test("it does not print title", () => {
  render(<Stepper steps={[stepTenureDetails, stepSelectResidents]} />);
  expect(screen.queryByText(stepperTitle)).not.toBeInTheDocument();
});
