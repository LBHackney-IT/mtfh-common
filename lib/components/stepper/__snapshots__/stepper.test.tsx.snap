// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders correctly 1`] = `
<div>
  <div>
    <h2
      class="lbh-heading-h2"
    >
      Stepper title
    </h2>
    <div
      class="mtfh-stepper mtfh-stepper--large mtfh-stepper--active"
    >
      <ol
        class="mtfh-stepper"
      >
        <li
          class="mtfh-stepper__step mtfh-stepper__step--active"
        >
          <div
            class="mtfh-stepper__header"
          >
            <h3
              class="lbh-heading-h3 mtfh-stepper__title"
            >
              <span
                class="mtfh-stepper__circle mtfh-stepper__circle--number"
              >
                <span
                  class="mtfh-stepper__circle-inner"
                >
                  <span
                    class="mtfh-stepper__circle-background"
                  >
                    <span
                      class="govuk-visually-hidden"
                    >
                      Step
                    </span>
                    1
                  </span>
                </span>
              </span>
              <span>
                <button
                  class="mtfh-stepper__button mtfh-stepper__button--title"
                >
                  <span>
                    Tenure details
                  </span>
                </button>
              </span>
            </h3>
          </div>
        </li>
        <li
          class="mtfh-stepper__step"
        >
          <div
            class="mtfh-stepper__header"
          >
            <h3
              class="lbh-heading-h3 mtfh-stepper__title"
            >
              <span
                class="mtfh-stepper__circle mtfh-stepper__circle--number"
              >
                <span
                  class="mtfh-stepper__circle-inner"
                >
                  <span
                    class="mtfh-stepper__circle-background"
                  >
                    <span
                      class="govuk-visually-hidden"
                    >
                      Step
                    </span>
                    2
                  </span>
                </span>
              </span>
              <span>
                <button
                  class="mtfh-stepper__button mtfh-stepper__button--title"
                >
                  <span>
                    Select residents
                  </span>
                </button>
              </span>
            </h3>
          </div>
        </li>
      </ol>
    </div>
  </div>
</div>
`;
